language/move-lang/src/unit_test/plan_builder.rs

```diff
@@ -0,0 +1,297 @@
+                ]),
+            };
+            args
+        }
+        EA::Parameterized(nm, attributes) => {
+            assert!(
+                nm.value == known_attributes::TestingAttributes::TEST,
+                "ICE: We should only be parsing a raw test attribute"
+            );
+            args.extend(
+                attributes
+                    .iter()
+                    .flat_map(|attr| parse_test_attribute(context, attr)),
+            );
+            args
+        }
+    }
+}
+
+fn parse_failure_attribute(
+    context: &mut Context,
+    sp!(aloc, expected_attr): &E::Attribute,
+) -> Option<ExpectedFailure> {
+    use E::{AttributeValue_ as EAV, Attribute_ as EA, Value_ as EV};
+    match expected_attr {
+        EA::Name(nm) => {
+            assert!(
+                nm.value == known_attributes::TestingAttributes::EXPECTED_FAILURE,
+                "ICE: We should only be parsing a raw expected failure attribute"
+            );
+            Some(ExpectedFailure::Expected)
+        }
+        EA::Assigned(_, sp!(assign_loc, _)) => {
+            context.env.add_error(vec![
+                (
+                    *assign_loc,
+                    "Invalid expected failure code assignment".into(),
+                ),
+                (
+                    *aloc,
+                    format!(
+                        "Expect an #[expected_failure({}=...)] attribute for abort code assignment",
+                        known_attributes::TestingAttributes::CODE_ASSIGNMENT_NAME
+                    ),
+                ),
+            ]);
+            None
+        }
+        EA::Parameterized(sp!(_, nm), attrs) => {
+            if attrs.len() != 1 {
+                context.env.add_error(vec![
+                    (*aloc, format!("Invalid #[expected_failure(...)] attribute, expected 1 argument but found {}", attrs.len())),
+                ]);
+                return None;
+            }
+            assert!(
+                nm == known_attributes::TestingAttributes::EXPECTED_FAILURE,
+                "ICE: expected failure attribute must have the right name"
+            );
+            match attrs.last().unwrap() {
+                sp!(_, EA::Assigned(sp!(_, nm), value))
+                    if nm == known_attributes::TestingAttributes::CODE_ASSIGNMENT_NAME =>
+                {
+                    match value {
+                        sp!(_, EAV::Value(sp!(_, EV::U64(u)))) => {
+                            Some(ExpectedFailure::ExpectedWithCode(*u))
+                        }
+                        sp!(_, EAV::Value(sp!(_, EV::U8(u)))) => {
+                            Some(ExpectedFailure::ExpectedWithCode(*u as u64))
+                        }
+                        sp!(_, EAV::Value(sp!(_, EV::U128(u)))) if *u <= std::u64::MAX as u128 => {
+                            Some(ExpectedFailure::ExpectedWithCode(*u as u64))
+                        }
+                        sp!(vloc, _) => {
```

Comment:
I think you need the InferredNum case here too

Reply:
Had a look at the `Value` definition and don't see it, so I don't think we need that here? But I may be wrong/missing something...